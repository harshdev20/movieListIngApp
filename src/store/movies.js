import data from "../api/CONTENTLISTINGPAGE-PAGE1.json";
import data1 from "../api/CONTENTLISTINGPAGE-PAGE2.json";
import data2 from "../api/CONTENTLISTINGPAGE-PAGE3.json";

//action types
const FETCH_ALL = "FETCH_ALL";
const SEARCH = "SEARCH";

let movies = data.page["content-items"].content;
let pageRequested = "";

//movie action
export function fetchMovies(page) {
  if (page === 1) {
    movies = data.page["content-items"].content;
    pageRequested = data.page["page-num-requested"];
  } else if (page === 2) {
    movies = data1.page["content-items"].content;
    pageRequested = data1.page["page-num-requested"];
  } else if (page === 3) {
    movies = data2.page["content-items"].content;
    pageRequested = data2.page["page-num-requested"];
  }
  return { type: FETCH_ALL, payload: { movies, pageRequested } };
}

//search action
export function search(key) {
  return { type: SEARCH, payload: key };
}

//initial state
const initialState = {
  movies: [],
  pageRequested: "",
  searchText: "",
};

//reducer
export const rootReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_ALL:
      return {
        ...state,
        movies: state.movies.concat(action.payload.movies),
        pageRequested: action.payload.pageRequested,
      };

    case SEARCH:
      return {
        ...state,
        searchText: action.payload,
      };

    default:
      return state;
  }
};
