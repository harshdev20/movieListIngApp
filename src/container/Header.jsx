import React, { Component } from "react";
import { connect } from "react-redux";
import { search } from "../store/movies";
import SearchBox from "../components/SearchBox";

class Header extends Component {
  render() {
    return (
      <div
        className="bg-transparent flex px-3 justify-between fixed w-full z-40 top-0 "
        style={{
          background: `url(${process.env.PUBLIC_URL + "/slices/nav_bar.png"} )`,
          backgroundPosition: "center",
          backgroundSize: "cover",
          backgroundRepeat: "no-repeat",
        }}
      >
        <div className="flex justify-center items-center">
          <img
            src={process.env.PUBLIC_URL + "/slices/Back.png"}
            alt="back_Button"
            className="w-5"
          />
          <p className="text-white px-5 text-sm ">Romantic Comedy</p>
        </div>
        <SearchBox onSearchChange={(e) => this.props.search(e.target.value)} />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    search: (key) => dispatch(search(key)),
  };
};
export default connect(null, mapDispatchToProps)(Header);
