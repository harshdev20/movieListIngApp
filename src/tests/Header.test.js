import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { create, act } from "react-test-renderer";
import Header from "../container/Header";
import { rootReducer } from "../store/movies";

const store = createStore(rootReducer);

it("should snapshot test Header", () => {
  let root;
  act(() => {
    root = create(
      <Provider store={store}>
        <Header />
      </Provider>
    );
  });

  expect(root.toJSON()).toMatchSnapshot();
});
