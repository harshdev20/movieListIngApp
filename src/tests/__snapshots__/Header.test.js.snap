// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot test Header 1`] = `
<div
  className="bg-transparent flex px-3 justify-between fixed w-full z-40 top-0 "
  style={
    Object {
      "background": "url(/slices/nav_bar.png )",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="flex justify-center items-center"
  >
    <img
      alt="back_Button"
      className="w-5"
      src="/slices/Back.png"
    />
    <p
      className="text-white px-5 text-sm "
    >
      Romantic Comedy
    </p>
  </div>
  <div
    className="relative flex w-1/4  flex-wrap items-end justify-end mb-3"
  >
    <input
      className="px-3 py-3 placeholder-gray-400 text-gray-700 relative rounded text-sm z-10 outline-none focus:outline-none focus:shadow-outline w-full pr-10 bg-black bg-opacity-0"
      onChange={[Function]}
      type="text"
    />
    <span
      className="h-full leading-snug font-normal absolute text-center text-gray-400 absolute rounded text-base items-center justify-center w-8 right-0 pr-3 py-3"
    >
      <img
        alt="back_Button"
        className="w-5"
        src="/slices/search.png"
      />
    </span>
  </div>
</div>
`;
